[jpa] Do not log failure to persist item with duplicate timestamp as error

ccutrer opened the pull request:
in case the user manually added a UNIQUE constraint to the database, openHAB might send duplicate timestamps.

effectively this means the first attempt is kept, while others are dropped. as long as you're using sub-second timestamps, this shouldn't be an issue - the state updates truly should be duplicates

jlaur commented on bundles/org.openhab.persistence.jpa/src/main/java/org/openhab/persistence/jpa/internal/JpaPersistenceService.java:
```diff
@@ -164,7 +165,12 @@ public void store(Item item, @Nullable String alias) {
         } catch (Exception e) {
-            logger.error("Error while persisting item! Rolling back!", e);
+            if (e.getCause() instanceof EntityExistsException) {
+                // there's a UNIQUE constraint in the database, and we tried to write
+                // a duplicate timestamp. Just ignore
```
Perhaps this should still be logged at log level **debug**? Since a value is actually discarded, this is still some kind of error situation, and it might be good to have the opportunity to backtrack this.

jlaur reviewed (changes requested):
Thanks! I have only one minor comment.

ccutrer commented:
@jlaur : anything else?

jlaur commented:
> anything else?

I don't think so, only the comment already posted. I don't know if I missed anything, but I only get notifications on commits and comments, not when a comment is only marked as resolved.

ccutrer commented:
🤦 🤦 🤦 

My bad. I _had_ addressed your comment locally. And even tried to push. But re-pushing just barely I realize for some reason my local branch didn't have its upstream set, so it just gave an error, and I hadn't noticed. This makes so much more sense why there has been no action! I apologize!

jlaur commented on bundles/org.openhab.persistence.jpa/src/main/java/org/openhab/persistence/jpa/internal/JpaPersistenceService.java:
```diff
@@ -164,7 +165,13 @@ public void store(Item item, @Nullable String alias) {
+            if (e.getCause() instanceof EntityExistsException) {
+                // there's a UNIQUE constraint in the database, and we tried to write
+                // a duplicate timestamp. Just ignore
+                logger.debug("Skipping persistence of item... it already exists");
```
For this to be of much use, I think we need to either add the item name, or include the call stack and hope the exception has enough details. E.g.:
```suggestion
                logger.debug("Failed to persist item {} because of duplicate timestamp", name);
```

jlaur commented:
> My bad. I _had_ addressed your comment locally. And even tried to push. But re-pushing just barely I realize for some reason my local branch didn't have its upstream set, so it just gave an error, and I hadn't noticed. This makes so much more sense why there has been no action! I apologize!

No worries! Next time please don't hestitate to ping me if you think I've forgotten about your PR - it happens. 😉

ccutrer commented:
Ping @jlaur : I think I addressed your comments.

jlaur commented:
> Ping @jlaur : I think I addressed your comments.

Did you see [this](https://github.com/openhab/openhab-addons/pull/15978#discussion_r1448039802)?

jlaur reviewed (approved):
LGTM

ccutrer commented:
Thanks! Sorry again for the false starts!
